File: Exercism/fsharp/prime-factors/PrimeFactors.fs
module PrimeFactors

let factors =
    let rec factors' candidate fs =
        function
        | n when n < 2L -> fs
        | n when n % candidate = 0L -> factors' candidate (candidate :: fs) (n / candidate)
        | n -> factors' (candidate + 1L) fs n

    factors' 2L [] >> List.rev >> List.map int

File: Exercism/csharp/prime-factors/PrimeFactors.cs
using System.Collections.Generic;

public static class PrimeFactors
{
    public static long[] Factors(long number)
    {
        var factor = 2;
        var factors = new List<long>();

        while (number > 1)
        {
            while (number % factor != 0) factor++;
            number /= factor;
            factors.Add(factor);
        }

        return factors.ToArray();
    }
}